library/java/org/cronvoy/CronvoyEngine.java

```diff
@@ -0,0 +1,229 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.cronvoy;
+
+import static android.os.Process.THREAD_PRIORITY_BACKGROUND;
+import static android.os.Process.THREAD_PRIORITY_MORE_FAVORABLE;
+
+import io.envoyproxy.envoymobile.EngineBuilder;
+import java.io.IOException;
+import java.net.Proxy;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.net.URLStreamHandlerFactory;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.chromium.net.BidirectionalStream;
+import org.chromium.net.ExperimentalBidirectionalStream;
+import org.chromium.net.NetworkQualityRttListener;
+import org.chromium.net.NetworkQualityThroughputListener;
+import org.chromium.net.RequestFinishedInfo;
+import org.chromium.net.UrlRequest;
+
+/**
+ * {@link java.net.HttpURLConnection} backed CronetEngine.
+ *
+ * <p>Does not support netlogs, transferred data measurement, bidistream, cache, or priority.
+ */
+public final class CronvoyEngine extends CronetEngineBase {
+
+  private final String mUserAgent;
+  private final ExecutorService mExecutorService;
+  private final io.envoyproxy.envoymobile.Engine engine;
+
+  public CronvoyEngine(CronetEngineBuilderImpl builder) {
+    // On android, all background threads (and all threads that are part
+    // of background processes) are put in a cgroup that is allowed to
+    // consume up to 5% of CPU - these worker threads spend the vast
+    // majority of their time waiting on I/O, so making them contend with
+    // background applications for a slice of CPU doesn't make much sense.
+    // We want to hurry up and get idle.
+    final int threadPriority =
+        builder.threadPriority(THREAD_PRIORITY_BACKGROUND + THREAD_PRIORITY_MORE_FAVORABLE);
+    this.mUserAgent = builder.getUserAgent();
+    EngineBuilder engineBuilder = new EngineBuilder();
+    engine = engineBuilder.build();
```

Comment:
```suggestion
    engine = engineBuilder.addLogLevel(LogLevel.TRACE).build();
```

Let's use trace logs and see what's going on internally